plugin/output/s3/s3.go

```diff
@@ -262,3 +290,122 @@ func (p *Plugin) generateObjectName(name string) string {
 	objectName = objectName[0 : len(objectName)-len(p.compressor.getExtension())]
 	return fmt.Sprintf("%s.%s%s", objectName, n, p.compressor.getExtension())
 }
+
+func (p *Plugin) createMinioClients() (minioClients map[string]objectStoreClient) {
+	minioClients = make(map[string]objectStoreClient)
+	// initialize minio clients object for main bucket.
+	mainClient, err := minio.New(p.config.Endpoint, p.config.AccessKey, p.config.SecretKey, p.config.Secure)
+	if err != nil || mainClient == nil {
+		p.logger.Panicf("could not create minio client, error: %s", err.Error())
```

Comment:
panic will be in case of nil err

Reply:
client can't be `nil`, I removed this condition at all.